### PR TITLE
Update utils package and rename dispose fn

### PR DESCRIPTION
**Pull request by mikesposito**
## Description

This PR pulls the latest version of `@metamask/utils`, which also includes an updated `Keyring` type with an optional `destroy` method: this allows us to remove some type narrowing and the ambiguity between `dispose` and `destroy` methods. 

## Changes

- **CHANGED**: `dispose` optional method is not called anymore when destroying a keyring

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by mcmire — APPROVED** [review]
Looks good!

**Review by Gudahtt — CHANGES_REQUESTED** [review]
I think we need to bump the minimum Node.js version first, this package still supports v14

File: package.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
   },
   "packageManager": "yarn@3.2.1",
   "engines": {
-    "node": ">=14.0.0"
+    "node": ">=16.0.0"
```
Comment (Gudahtt) [inline package.json:84]: Thoughts on moving this to a separate PR? It would be easier to keep track of that way

Comment (mikesposito) [reply package.json:84]: Yes, I think it would be better to separate the scopes into two PRs

I created [236](https://github.com/MetaMask/KeyringController/pull/236), that we could merge before this PR

**Review by Gudahtt — APPROVED** [review]
LGTM!
